pkg/subctl/resource/schedule_pod.go

```diff
@@ -0,0 +1,186 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package resource
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type NetworkPodType int
+
+type NetworkPodScheduling int
+
+const (
+	GatewayLabel = "submariner.io/gateway"
+)
+
+const (
+	InvalidScheduling NetworkPodScheduling = iota
+	GatewayNode
+	NonGatewayNode
+)
+
+type NetworkingType bool
+
+const (
+	HostNetworking NetworkingType = true
+	PodNetworking  NetworkingType = false
+)
+
+type PodConfig struct {
+	Name       string
+	ClientSet  *kubernetes.Clientset
+	Scheduling NetworkPodScheduling
+	Networking NetworkingType
+	Namespace  string
+	Command    string
+	Timeout    uint
+}
+
+type NetworkPod struct {
+	Pod       *v1.Pod
+	Config    *PodConfig
+	PodOutput string
+}
+
```

Comment:
We should be able to reuse at least some of this from shipyard framework. I can look into that.

Reply:
I tried to reuse wherever possible, but if you find anything please suggest. Thanks.

Reply:
I created https://github.com/submariner-io/shipyard/issues/493